### PR TITLE
[WIP] Recipe for Popt

### PR DESCRIPTION
**Pull request by aviks**
Doesn't work on Windows yet.

### CODE REVIEW

**Comment by giordano** [comment]
https://github.com/msys2/MINGW-packages/tree/4be00b98f6cd258d1665c7787ddef4d1e7c5d7b4/mingw-w64-popt might help

**Comment by aviks** [comment]
Thanks! That is a *ton* of patches 😢

**Comment by giordano** [comment]
But if you're lucky you just need to apply them and that's it :grin:

File: P/Popt/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+atomic_patch -p1 "${WORKSPACE}/srcdir/patches/318833.all.patch"
+atomic_patch -p1 "${WORKSPACE}/srcdir/patches/356669.all.patch"
+atomic_patch -p1 "${WORKSPACE}/srcdir/patches/367153-manpage.all.patch"
+atomic_patch -p1 "${WORKSPACE}/srcdir/patches/get-w32-console-maxcols.mingw32.patch"
+atomic_patch -p1 "${WORKSPACE}/srcdir/patches/no-uid-stuff-on.mingw32.patch"
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --disable-static
+make -j${nproc}
+make install
+install_license COPYING
+false
```
Comment (giordano) [inline P/Popt/build_tarballs.jl:32]: I guess this was for your local testing? :smiley: 
```suggestion
```

**Comment by giordano** [comment]
When a builder fails for PowerPC and FreeBSD because the shared library hadn't been compiled, 99.99% of the times it's because of an old `configure` script: https://github.com/JuliaPackaging/Yggdrasil/wiki/Build-Tricks#shared-library-not-built

**Comment by aviks** [comment]
Ah, yes, I knew there was something, but couldnt find the docs. I looked at BB docs and BB wiki :(

**Comment by aviks** [comment]
Ah, well, doesnt seem to have helped 😦

**Comment by giordano** [comment]
Well, the problem is that the source code is _old_ and it's using very old macros in `configure.ac` or the M4 files.  I might be able to have a look later tonight.

**Comment by aviks** [comment]
Thanks, if you can, but I'm happy to remove ppc and freebsd from the supported list for this.

**Comment by aviks** [comment]
Thanks a ton, Mose.

**Pull request merged**